source/core/CodalComponent.cpp

```diff
@@ -33,6 +33,10 @@ CodalComponent* CodalComponent::components[DEVICE_COMPONENT_COUNT];
 
 uint8_t CodalComponent::configuration = 0;
 
+#if DEVICE_COMPONENT_COUNT > 255
+#error "DEVICE_COMPONENT_COUNT has to fit in uint8_t"
```

Comment:
this is for existing code that uses uint8 for some reason

Reply:
Weird!

Reply:
maybe we should just put the components on a linked list and pre-allocate an array for them. Would make iterating backwards harder, but I guess not too bad. Well, in future, when we have time...